clang/test/Driver/hip-wavefront-size-deprecation-diagnostics.hip

```diff
@@ -0,0 +1,111 @@
+// REQUIRES: amdgpu-registered-target
+// RUN: %clang -xhip --offload-arch=gfx1030 --offload-host-only -pedantic -nogpuinc -nogpulib -nobuiltininc -fsyntax-only -Xclang -verify %s
+// RUN: %clang -xhip --offload-arch=gfx1030 --offload-device-only -pedantic -nogpuinc -nogpulib -nobuiltininc -fsyntax-only -Xclang -verify %s
+
+// Test that deprecation warnings for the wavefront size macro are emitted properly.
+
+#include <type_traits>
```

Comment:
1. Clang's tests must be hermetic and can't include system headers
2. This fails on Mac: http://45.33.8.238/macm1/95266/step_6.txt

Please take a look and revert for now if it takes a while to fix.

Reply:
@nico Thanks for the report and sorry for the inconvenience! I reverted the patch with #115499 and opened PR #115507 with a fix. I'd appreciate it if you could take a look and see if that fixes the issue.

Reply:
Thanks! Since the test is now hermetic, I'm guessing it'll work now. I'll check my bot after you've relanded your patch.